### ISSUE TITLE
Cannot install dotnet-codegen

### ISSUE THREAD

**Issue by Nihlus** [AArnott/CodeGeneration.Roslyn#196]
When trying to create a new minimal project with 0.7.5-alpha, the installation of dotnet-codegen fails with a NU1212 error. The reason for this is, as far as I can tell, that the package type is explicitly specified in https://github.com/AArnott/CodeGeneration.Roslyn/blob/master/src/CodeGeneration.Roslyn.Tool/CodeGeneration.Roslyn.Tool.csproj#L4 as `DotnetCliTool`, whereas the most recent version of `dotnet` expects `DotnetTool`.

A possible workaround or solution would be to drop the explicit package type specification, and use `<PackAsTool>true</PackAsTool>` instead (as seen in https://docs.microsoft.com/en-us/dotnet/core/tools/global-tools-how-to-create).

```
jarl@nessie ~ $ dotnet --version
3.1.102
jarl@nessie ~ $ dotnet --list-sdks
2.1.804 [/usr/share/dotnet/sdk]
3.1.102 [/usr/share/dotnet/sdk]
jarl@nessie ~ $ dotnet --list-runtimes
Microsoft.AspNetCore.All 2.1.16 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
Microsoft.AspNetCore.App 2.1.16 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.2 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 2.1.16 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.2 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

**Comment by amis92** [comment]
@Nihlus and what did you attempt as a command for installing `dotnet-codegen`?

Unfortunately, there's no command-line command to do that, you have to manually edit the project file.

Also, while you're at it, 0.7-alpha is unusable as seen in #195

**Comment by Nihlus** [comment]
@amis92 No command at first, just letting dotnet restore figure things out. After that failed, I ended up trying to install it as a global tool, which didn't work for the same but more verbose reason.

Strangely, it all started working for no apparent reason a little earlier today. I tried adding the tool with `dotnet add package` first, and then changing it to a tool reference - apparently that jogged something, and the generators are working fine now.
